frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -322,6 +310,21 @@ impl BenchmarkingConfig for () {
 	const MAXIMUM_TARGETS: u32 = 2_000;
 }
 
+/// A fallback implementation that transitions the pallet to the emergency phase.
+pub struct InitiateEmergencyPhase<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> ElectionProvider<T::AccountId, T::BlockNumber> for InitiateEmergencyPhase<T> {
+	type DataProvider = T::DataProvider;
+	type Error = &'static str;
+
+	fn elect() -> Result<Supports<T::AccountId>, Self::Error> {
+		log!(warn, "Entering emergency phase.");
+		CurrentPhase::<T>::put(Phase::Emergency);
```

Comment:
the implementation of ElectionProvider on Pallet already set to emergency phase when the election fail, should that be removed ?
Like if there is an on-chain fallback, we probably don't want to enter the emergency phase.

Reply:
not sure if I expressed myself clearly, but I think we should remove the set of emergency phase inside the implementation on `Pallet` not here

Reply:
ah indeed I missread, the elect will call the fallback, so if the fallback gives an error, then we move to emergency phase.

So yes I agree, better to have InitiateEmergency just fail, and always go to emergency in case the fallback fails